### ISSUE TITLE
Dockerfile.devel-gpu: infinite prompt loop

### ISSUE THREAD

**Issue by flx42** [tensorflow/tensorflow#15291]
```sh
$ docker build -f Dockerfile.devel-gpu github.com/tensorflow/tensorflow.git#master:tensorflow/tools/docker
[...]
Step 19/22 : RUN ln -s /usr/local/cuda/lib64/stubs/libcuda.so /usr/local/cuda/lib64/stubs/libcuda.so.1 &&     LD_LIBRARY_PATH=/usr/local/cuda/lib64/stubs:${LD_LIBRARY_PATH}     tensorflow/tools/ci_build/builds/configured GPU     bazel build -c opt --config=cuda --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=0"         tensorflow/tools/pip_package:build_pip_package &&     rm /usr/local/cuda/lib64/stubs/libcuda.so.1 &&     bazel-bin/tensorflow/tools/pip_package/build_pip_package /tmp/pip &&     pip --no-cache-dir install --upgrade /tmp/pip/tensorflow-*.whl &&     rm -rf /tmp/pip &&     rm -rf /root/.cache
 ---> Running in 3e8560e2d441
/tensorflow /tensorflow
INFO: Reading 'startup' options from /etc/bazel.bazelrc: --batch
Extracting Bazel installation...
You have bazel 0.5.4 installed.
Found possible Python library paths:
  /usr/local/lib/python2.7/dist-packages
  /usr/lib/python2.7/dist-packages
Please input the desired Python library path to use.  Default is [/usr/local/lib/python2.7/dist-packages]
Do you wish to build TensorFlow with jemalloc as malloc support? [Y/n]: jemalloc as malloc support will be enabled for TensorFlow.

Do you wish to build TensorFlow with Google Cloud Platform support? [Y/n]: Google Cloud Platform support will be enabled for TensorFlow.

Do you wish to build TensorFlow with Hadoop File System support? [Y/n]: Hadoop File System support will be enabled for TensorFlow.

Do you wish to build TensorFlow with Amazon S3 File System support? [Y/n]: Amazon S3 File System support will be enabled for TensorFlow.

Do you wish to build TensorFlow with XLA JIT support? [y/N]: No XLA JIT support will be enabled for TensorFlow.

Do you wish to build TensorFlow with GDR support? [y/N]: No GDR support will be enabled for TensorFlow.

Do you wish to build TensorFlow with VERBS support? [y/N]: No VERBS support will be enabled for TensorFlow.

Do you wish to build TensorFlow with OpenCL support? [y/N]: No OpenCL support will be enabled for TensorFlow.

Please specify the CUDA SDK version you want to use, e.g. 7.0. [Leave empty to default to CUDA 8.0]: 

Please specify the location where CUDA 8.0 toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 

Invalid path to CUDA 8.0 toolkit. /usr/local/cuda/lib64/libcudart.so.8.0 cannot be found
Please specify the CUDA SDK version you want to use, e.g. 7.0. [Leave empty to default to CUDA 8.0]: 

Please specify the location where CUDA 8.0 toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 

Invalid path to CUDA 8.0 toolkit. /usr/local/cuda/lib64/libcudart.so.8.0 cannot be found
Please specify the CUDA SDK version you want to use, e.g. 7.0. [Leave empty to default to CUDA 8.0]: 

Please specify the location where CUDA 8.0 toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 

Invalid path to CUDA 8.0 toolkit. /usr/local/cuda/lib64/libcudart.so.8.0 cannot be found
Please specify the CUDA SDK version you want to use, e.g. 7.0. [Leave empty to default to CUDA 8.0]: 

Please specify the location where CUDA 8.0 toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 

Invalid path to CUDA 8.0 toolkit. /usr/local/cuda/lib64/libcudart.so.8.0 cannot be found
Please specify the CUDA SDK version you want to use, e.g. 7.0. [Leave empty to default to CUDA 8.0]: 

Please specify the location where CUDA 8.0 toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 

Invalid path to CUDA 8.0 toolkit. /usr/local/cuda/lib64/libcudart.so.8.0 cannot be found
Please specify the CUDA SDK version you want to use, e.g. 7.0. [Leave empty to default to CUDA 8.0]: 

Please specify the location where CUDA 8.0 toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 

Invalid path to CUDA 8.0 toolkit. /usr/local/cuda/lib64/libcudart.so.8.0 cannot be found
Please specify the CUDA SDK version you want to use, e.g. 7.0. [Leave empty to default to CUDA 8.0]: 

Please specify the location where CUDA 8.0 toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 

[... ad infinitum ... ]
```

Is it because the `1.4` branch doesn't support CUDA 9.0?
https://github.com/tensorflow/tensorflow/blob/abd5375ba8d373045321d1eebdb4501c36ab0ccd/tensorflow/tools/docker/Dockerfile.devel-gpu#L74-L76

@gunan

**Comment by gunan** [comment]
The reason is 1.4 branch defaults to CUDA 8, and if you do not have cuda8 available on the image.
@angersson actually fixed this internally to avoid the infinite loop there.

You should be able to get around this by setting TF_CUDA_VERSION to 9.0, and TF_CUDNN_VERSION to 7 in the image.
In the future, we should have this controlled by ARGS in the image.

**Comment by flx42** [comment]
> You should be able to get around this by setting TF_CUDA_VERSION to 9.0, and TF_CUDNN_VERSION to 7 in the image.

Sure, but I'm reporting that your official Dockerfile is broken ;)

**Comment by angersson** [comment]
Actually, I think the CUDA prompt is one of the ones I didn't update because of its complexity (and potential breakages due to change).

I'm working on replicating this.

**Comment by angersson** [comment]
Worth noting: the [README in the Docker directory](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/docker) says not to build the containers directly, though that may be out of date.

**Comment by flx42** [comment]
@angersson As far as I know, the templating only happens for the `runtime` Dockerfiles (`Dockerfile` and `Dockerfile.gpu`). I don't think `Dockerfile.devel-gpu` is modified in any way.

**Comment by flx42** [comment]
Also, I'm probably not smart enough to fathom how `parameterized_docker_build.sh` works :)

**Comment by angersson** [comment]
This has been fixed by #15328.
